ouroboros-consensus/src/Ouroboros/Consensus/Ledger/Query.hs

```diff
@@ -162,16 +228,26 @@ queryDecodeNodeToClient codecConfig queryVersion blockVersion
         blockVersion
       return (SomeSecond (BlockQuery blockQuery))
 
-instance SerialiseResult blk (BlockQuery blk) => SerialiseResult blk (Query blk) where
+instance  ( SerialiseResult blk (BlockQuery blk)
+          , Serialise (HeaderHash blk)
```

Comment:
We have a general preference for `ToCBOR` and `FromCBOR` instead of `Serialise`. But using `Serialise` here is not without precedent (eg `BlockQuery ShelleyBlock` uses `Serialise` for a `Point`'s `HeaderHash`).

Ultimately, the `Serialise` vs `ToCBOR`/`FromCBOR` vs `SerialiseNodeToClient`/`SerialiseNodeToNode` vs `SerialiseResult` is in a confused and confusing state, in my opinion. The only stark contrast is that many "low-level" `Serialise` instances come from 3rd party libraries, and so we shouldn't be using them: changes in our codecs should never be accidental, and so shouldn't run the risk of accidentally depending on 3rd party serializations.

So: if it's not too much trouble, I would prefer at least `ToCBOR` and `FromCBOR` instead of `Serialise`. But really, this is already in the scope of existing tech debt :(

Reply:
If it's okay I'd like to do this in a separate PR.